odl/contrib/fom/supervised/supervised_fom.py

```diff
@@ -0,0 +1,570 @@
+                             {\| f - \\overline{f} \|_2 +
+                              \| g - \\overline{g} \|_2 } \\bigg \\rvert,
+
+    where
+
+    .. math::
+        \\overline{f} := \\frac{1}{\|1_\Omega\|_1} \\int_\Omega f dx,
+
+    and
+
+    .. math::
+        \\overline{g} := \\frac{1}{\|1_\Omega\|_1} \\int_\Omega g dx.
+
+    The normalized FOM takes its values in [0, 1], with higher correspondance
+    at lower FOM value.
+
+    Parameters
+    ----------
+    test_data : `LinearSpaceElement`
+        Input data or reconstruction.
+    ref_data : `LinearSpaceElement`
+        Reference data to compare ``test_data`` to.
+    mask : `LinearSpaceElement`
+        Binary mask to define ROI in which FOM evaluation is performed.
+    normalized  : `Boolean`
+        Boolean flag to switch between unormalized and normalized FOM.
+
+    Returns
+    -------
+    fom : `Scalar`
+        Scalar (float) indicating absolute difference in standard deviation
+        between ``test_data`` and ``ref_data``. In normalized form the FOM
+        takes values in [0, 1], with higher correspondance at lower FOM value.
+    """
+
+    import numpy as np
+    import odl
+
+    l1_norm = odl.solvers.L1Norm(test_data.space)
+    l2_norm = odl.solvers.L2Norm(test_data.space)
+
+    if mask:
+        test_data = test_data * mask
+        ref_data = ref_data * mask
+
+    # Identify positive part of test_data
+    pos_test_data = test_data.copy()
+    pos_test_data = pos_test_data.asarray()
+    pos_test_data[pos_test_data < 0] = 0
+    pos_test_data = test_data.space.element(pos_test_data)
+
+    # Identify negative part of test_data
+    neg_test_data = test_data.copy()
+    neg_test_data = neg_test_data.asarray()
+    neg_test_data[neg_test_data > 0] = 0
+    neg_test_data = test_data.space.element(neg_test_data)
+
+    # Identify positive part of ref_data
+    pos_ref_data = ref_data.copy()
+    pos_ref_data = pos_ref_data.asarray()
+    pos_ref_data[pos_ref_data < 0] = 0
+    pos_ref_data = ref_data.space.element(pos_ref_data)
+
+    # Identify negative part of ref_data
+    neg_ref_data = ref_data.copy()
+    neg_ref_data = neg_ref_data.asarray()
+    neg_ref_data[neg_ref_data > 0] = 0
+    neg_ref_data = ref_data.space.element(neg_ref_data)
+
+    # Volume of space
+    vol = l1_norm(test_data.space.one())
+
+    # Mean value of test_data
+    test_data_mean = (1 / vol) * (l1_norm(pos_test_data) -
+                                  l1_norm(neg_test_data))
+
+    # Mean value of ref_data
+    ref_data_mean = (1 / vol) * (l1_norm(pos_ref_data) -
+                                 l1_norm(neg_ref_data))
+
+    fom = np.abs((l2_norm(test_data - test_data_mean) -
+                  l2_norm(ref_data - ref_data_mean)))
+
+    if normalized:
+        fom /= (l2_norm(test_data - test_data_mean) +
+                l2_norm(ref_data - ref_data_mean))
+
+    return fom
+
+
+def range_difference(test_data, ref_data, mask=None, normalized=False):
```

Comment:
I am not sure about this, perhaps someone could help clear this out. Wikipedia says the following about "dynamic range":

> It is measured either as a ratio or as a base-10 (decibel) or base-2 (doublings, bits or stops) logarithmic value of the difference between the smallest and largest signal values